cmake/public/cuda.cmake

```diff
@@ -89,6 +94,10 @@ find_library(CUDA_NVRTC_LIB nvrtc
     PATH_SUFFIXES lib lib64 lib/x64)
 
 # Create new style imported libraries.
+# Several of these libraries have a hardcoded path if CAFFE2_STATIC_LINK_CUDA
+# is set. This path is where sane CUDA installations have their static
+# libraries installed. Users may have strange CUDA installations, but they
+# should never have this flag set and so shouldn't run into problems.
```

Comment:
Nit: maybe say "this flag should only be used for binary builds, so end users should never have this flag set"